apps/dashboard/app/controllers/projects_controller.rb

```diff
@@ -110,9 +110,34 @@ def job_details
 
     hpc_job = project.job(job_details_params[:jobid].to_s, cluster_str)
 
+    @project = project
     render(partial: 'job_details', locals: { job: hpc_job })
   end
 
+  # DELETE /projects/:project_id/jobs/:cluster/:jobid
+  def delete_job
+    project = Project.find(job_details_params[:project_id])
+    cluster_str = job_details_params[:cluster].to_s
+    hpc_job = project.job(job_details_params[:jobid].to_s, cluster_str)
+
+    if hpc_job.status.to_s == 'completed'
+      redirect_to project_path(job_details_params[:project_id]) if hpc_job.status.to_s == 'completed'
+    else
+      project.remove_logged_job(job_details_params[:jobid].to_s, cluster_str)
+      redirect_to project_path(job_details_params[:project_id])
+    end
```

Comment:
I think this logic is off. If the job is completed, this is where we provide the functionality to remove it from the job log.

If it's not completed, first the UI shouldn't have a delete button, but even if they try to delete a job that's not complete... We could either throw the error that they need to stop it first, or stop it and remove it. I'm inclined to think we should do the former just because it's more conservative.

Reply:
🤦 This logic was meant for stopping a job rather than deleting it, though throwing an error would be better than just returning for sure.